### PR TITLE
Create container.yml

### PR DESCRIPTION
**Pull request by findyoky**

### CODE REVIEW

File: .github/workflows/container.yml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    branches:
+      - main
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
```
Comment (elveskevtar) [inline .github/workflows/container.yml:14]: Does this actually run the jobs in the application container(s)?

Comment (elveskevtar) [reply .github/workflows/container.yml:14]: Ahhh so I did not understand what this was doing at first. It seems to be doing static code analysis, scanning relevant files for particular docker security issues.

My only comment is that these are pretty basic checks that could be made more helpful with some tweaking. But for now, let's merge it and we can continue to improve on these if we have time at the end.

File: .github/workflows/container.yml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+          if ! grep -q 'secrets:' **/*.yml; then
+            echo "Warning: Docker Secrets are not utilized."
+          fi
+
+      # Rule #13: Enhance Supply Chain Security
+      - name: Check for SBOM Generation
+        run: |
+          if ! grep -q 'sbom:' **/*.yml; then
+            echo "Warning: SBOM generation is not defined."
+          fi
```
Comment (elveskevtar) [inline .github/workflows/container.yml:111]: I don't think there is any sbom specification we can add to Dockerfile/Docker compose file. It looks like it's actually just a command line tool: https://docs.docker.com/scout/how-tos/view-create-sboms/

We can make a followup to add this.

File: .github/workflows/container.yml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+          if [ "$(docker info | grep 'rootless')" != "rootless: true" ]; then
+            echo "Warning: Docker is not running in rootless mode."
+          fi
+
+      # Rule #12: Utilize Docker Secrets
+      - name: Check for Docker Secrets Usage
+        run: |
+          if ! grep -q 'secrets:' **/*.yml; then
+            echo "Warning: Docker Secrets are not utilized."
+          fi
```
Comment (elveskevtar) [inline .github/workflows/container.yml:104]: I used docker secrets initially, but then decided on just using dotenv file. It is easier for our purposes and fine for local development (until we want to deploy to actual infrastructure)

File: .github/workflows/container.yml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            echo "Warning: No memory limits defined."
+          fi
+
+      # Rule #8: Set Filesystem and Volumes to Read-Only
+      - name: Check for Read-Only Filesystem
+        run: |
+          if grep -q 'read_only: false' **/*.yml; then
+            echo "Error: Read-only filesystem is not set."
+            exit 1
+          fi
```
Comment (elveskevtar) [inline .github/workflows/container.yml:78]: This is checking all yml files in the repo for the existence of read_only: false which I'm pretty sure we don't have. Even if we do, it does not guarantee that our docker container's filesystem has this set. We might want to comment this out for now and come back to it or do a deeper check.

**Review by wbc2025 — APPROVED** [review]
approved, though some of these checks may fail on our code

**Pull request merged**